kolibri/plugins/learn/assets/src/views/LearnIndex.vue

```diff
@@ -208,6 +223,21 @@
         return isAssessment ? BOTTOM_SPACED_RESERVED : 0;
       },
     },
+    watch: {
+      $route: function(newRoute, oldRoute) {
+        // Return if the current route is a search path. Otherwise this method creates
+        // an infinite loop.
+        if (newRoute.name === 'SEARCH' && oldRoute.name === 'SEARCH') {
+          return;
+        }
+
+        // Destructure the oldRoute into an object with the only four properties.
+        // Setting this.lastRoute = oldRoute causes issues for some reason.
+        this.lastRoute = (({ name, query, path, params }) => ({ name, query, path, params }))(
```

Comment:
Might be simpler to just create a new object from oldRoute. Like `lastRoute = { name: oldRoute.name }`, etc. And I don't think router-link objects need the `path` property. name, query, and params always are enough.